### PR TITLE
feat(alert): Moving alert builder into sidebar page

### PR DESCRIPTION
**Pull request by davidenwang**
Moves the alert builder into the sidebar item created from https://github.com/getsentry/sentry/pull/20020
New route lives under 
- `organizations/:orgId/alerts/:projectId/new`
- `.../alerts/metric-rules/:projectId` or `...alerts/rules/:projectId` for editing

Also created a new `builder/projectProvider.tsx` component which will handle fetching the project for the alert/metric-alert builders

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/details/index.tsx [hunk 1]
```diff
@@ -154,4 +153,4 @@ class IncidentDetails extends React.Component<Props, State> {
   }
 }
 
-export default withApi(withOrganization(IncidentDetails));
+export default withApi(IncidentDetails);
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/details/index.tsx:156]: removed this as the `AlertsContainer` now populates its children (`IncidentDetails`, `AlertRulesListContainer`) with an organization

File: src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx [hunk 2]
```diff
@@ -43,13 +46,17 @@ class Create extends React.Component<Props, State> {
...
+      location,
+      params: {projectId},
+    } = this.props;
 
     trackAnalyticsEvent({
       eventKey: 'new_alert_rule.viewed',
       eventName: 'New Alert Rule: Viewed',
       organization_id: organization.id,
-      project_id: project.id,
+      project_id: projectId,
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx:59]: project.id was a number projectId is the slug, probably didn't need to switch. Currently errors for me

**Review by scttcper — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx (hunk 2 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx:59]: ooh good catch changing it back now

**Review by scttcper — APPROVED** [review]
looks really good! 👍

File: src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx (hunk 2 above)
Comment (iProgramStuff) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx]: Also getting this error
`Error: Unable to coerce value to number: 'internal'`

Comment (iProgramStuff) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx]: Sorry, this error is gone. Tried `git pull` after seeing your other commits

**Review by iProgramStuff — APPROVED** [review]
Everything seems to check out to me.

**Pull request merged**